### PR TITLE
Make use of rsync optional in build.sh

### PR DESCRIPTION
**Pull request by ssbarnea**
Fixes: #306

### CODE REVIEW

**Review by felixfontein — CHANGES_REQUESTED** [review]
The whole reason of using rsync with the temp directory is that Sphinx only rebuilds the changed files in subsequent runs, and that files that should no longer be there are deleted. You cannot simply replace rsync by cp here without breaking both these things.

Instead of doing some silent fallback, I would implement this differently so that the user can pass a flag to `antsibull-docs sphinx-init` that directly lets antsibull-docs write to the destination instead of writing into a temp directory and using rsync. You can then use that option in CI when you start with a clean state anyway and don't need incremental builds.

**Comment by felixfontein** [comment]
(To re-create the sphinx-init baseline, go to `tests/functional/` and run `./build-sphinx-init-baseline.sh`.)

File: src/antsibull_docs/data/sphinx_init/build_sh.j2 [hunk 1]
```diff
@@ -4,12 +4,26 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 # Created with antsibull-docs @{ antsibull_docs_version }@
+# cspell: disable cprv
```
Comment (gotmax23) [inline src/antsibull_docs/data/sphinx_init/build_sh.j2:7]: We don't use cspell in this repository (whether or not we should is a separate discussion), so it'd be better not to have comments for it cluttering up the sources.

**Comment by felixfontein** [comment]
I'm planning to do a new feature release of antsibull-docs this weekend. This would be a nice feature to include; when do you plan to continue working on it?

**Comment by felixfontein** [comment]
#315 implements this differently.

**Pull request closed without merging**